### PR TITLE
Add product on FE update product route scraper error

### PR DESCRIPTION
**Pull request by surhud004**
#30 Done. Paste product url on both bars -> progress loading circle shows up -> scraper returns data -> stores in DB -> display product on FE.

### CODE REVIEW

**Comment by Ethorf** [comment]
Looking good so far!

File: client/public/index.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>BigDeal</title>
```
Comment (sina-jamshidi) [inline client/public/index.html:28]: Nice detail!

File: server/routes/products.js [hunk 2]
```diff
@@ -50,16 +50,23 @@ router.post('/lists/:id/products/new', verifyToken, async function (req, res) {
...
-				description: product.description,
-				url: url,
-				lastprice: 0.0,
+			let description = '';
+			if(typeof product.description === "string") {
+				description = product.description;
+			} else {
+				description = product.description[0];
+			}
+			var newProduct = {
```
Comment (sina-jamshidi) [inline server/routes/products.js]: Can we change this to `let`? We should generally avoid `var`

**Comment by surhud004** [comment]
Also updated a few Material UI Icons on buttons! Added loading progress circle on Lists bar too!!

**Comment by ChianHuei** [comment]
The loading progress indicator is a good idea.

**Review by ChianHuei — APPROVED** [review]

File: client/src/form/DropDownListBar.js [hunk 3]
```diff
@@ -57,14 +65,28 @@ export default function BasicTextFields() {
...
+                        listId: '',
+                        url: '',
+                    });
+                }
+            });
+        }
+        return () => {
+            isUnmounted = true;
+        };
+    }, [loading]);
```
Comment (sina-jamshidi) [inline client/src/form/DropDownListBar.js:88]: I'm a little confused about the `isUnmounted` cleanup logic. Can you explain the thinking here?

**Review by sina-jamshidi — COMMENTED** [review]
Looks good

File: client/src/form/DropDownListBar.js (hunk 3 above)
Comment (surhud004) [reply client/src/form/DropDownListBar.js:88]: So basically I was searching for the error on stackoverflow and reading how to use useEffects in React docs. So I found a code - https://codesandbox.io/s/use-local-variable-to-avoid-setstate-on-unmouted-component-6cmie where a similar logic of isCancelled and setMounted was added. I followed the code and wrote this and it worked somehow!!

Comment (surhud004) [reply client/src/form/DropDownListBar.js:88]: So as far as I understood, we are calling a promise onSubmit and also updating the state (setLoading) so we need to move this code to useEffect and check if the component is mounted or not (something called as clean up function), a temp flag which returns true if the component is unmounted before the promise completes. Is that right?!

**Pull request merged**